symmecrypt_test.go

```diff
@@ -450,3 +460,327 @@ func ExampleNewReader() {
 		panic(err)
 	}
 }
+
+func TestSequentialEncryption(t *testing.T) {
+	var content = "this is a very sensitive content"
+
+	// The key will be Instantiate from the sha152 of the content
+	hash := sha512.New512_256()
+	_, err := io.Copy(hash, strings.NewReader(content))
+	require.NoError(t, err)
+	sha512 := hex.EncodeToString(hash.Sum(nil))
+
+	// Prepare a keyloadConfig to be able to Instantiate the key properly
+	cfg := symmecrypt.KeyConfig{
+		Cipher:     aesgcm.CipherName,
+		Sequential: true,
+		Key:        sha512,
+	}
+
+	// Instantiate a Sequential key from the sha512
+	k, err := keyloader.NewKey(&cfg)
```

Comment:
we should encapsulate determinic stuff in a `deterministic` package

`deterministic.NewSequentialKey(...)`